src/utils/hooks/useListApplications.tsx

```diff
@@ -1,98 +1,43 @@
 import { useEffect, useState } from 'react'
-import {
-  Application,
-  ApplicationStageStatusAll,
-  useGetApplicationsQuery,
-  useGetApplicationsStagesQuery,
-} from '../../utils/generated/graphql'
-import { ApplicationDetails } from '../types'
+import buildFilter from '../helpers/application/buildQueryFilters'
+import buildSortFields, { getPaginationVariables } from '../helpers/application/buildQueryVariables'
+import { useGetApplicationsListQuery, ApplicationList } from '../../utils/generated/graphql'
+import { QueryObject } from '../hooks/useRouter'
 
-interface UseListApplicationsProps {
-  type?: string
+export type URLQueryFilter = {
+  [key: string]: string
 }
 
-interface ApplicationDetailsMap {
-  [serial: string]: ApplicationDetails
-}
-
-const useListApplications = ({ type }: UseListApplicationsProps) => {
-  const [applications, setApplications] = useState<ApplicationDetailsMap>()
+const useListApplications = (urlFilters: QueryObject) => {
+  const [applications, setApplications] = useState<ApplicationList[]>([])
   const [loading, setLoading] = useState(true)
   const [error, setError] = useState('')
+  const { type, sortBy, page, perPage } = urlFilters
```

Comment:
```suggestion
const useListApplications = ({ sortBy, page, perPage, ...queryFilters }: QueryObject) => {
  const [applications, setApplications] = useState<ApplicationList[]>([])
  const [loading, setLoading] = useState(true)
  const [error, setError] = useState('')
```

Reply:
But since we need `type` in both this main function and passed into `buildFilter` we still need to do `const { type } = queryFilters` (or refer to `queryFilters.type` later), so it doesn't end up looking cleaner or clearer I think.